contracts/transfer/circuits/src/dusk_contract/execute.rs

```diff
@@ -180,32 +277,72 @@ impl Circuit<'_> for ExecuteCircuit {
 
         // Iterate over the note elements and hash them together
         // and constrain against the hash of the note.
-        input_commitments
+        input_note_types
             .iter()
-            .zip(position_of_notes.iter())
+            .zip(input_commitments.iter())
+            .zip(input_nonces.iter())
             .zip(input_notes_pk.iter())
+            .zip(input_randomness.iter())
+            .zip(position_of_notes.iter())
+            .zip(input_ciphers_one.iter())
+            .zip(input_ciphers_two.iter())
+            .zip(input_ciphers_three.iter())
             .zip(note_hashes.iter())
-            .for_each(|(((value, position), key), note_hash)| {
-                let computed_hash = sponge_hash_gadget(
-                    composer,
-                    &[*value.x(), *value.y(), position.var, *key.x(), *key.y()],
+            .for_each(
+                |(
+                    (
+                        (
+                            (
+                                (
+                                    (
+                                        (
+                                            ((note_type, commitment), nonce),
+                                            pk_r,
+                                        ),
+                                        randomness,
+                                    ),
+                                    position,
+                                ),
+                                cipher_one,
+                            ),
+                            cipher_two,
+                        ),
+                        cipher_three,
+                    ),
+                    note_hash,
+                )| {
+                    let computed_hash = input_preimage(
+                        composer,
+                        *note_type,
+                        *commitment,
+                        *nonce,
+                        *pk_r,
+                        *randomness,
+                        *position,
+                        *cipher_one,
+                        *cipher_two,
+                        *cipher_three,
+                    );
+
+                    composer.assert_equal(computed_hash, note_hash.var);
```

Comment:
That's not correct AFAIK. You want to constrain the hash of the Note against a public input. Not against a private variable. Therefore, this should be changed.

Reply:
This is not correct, as per the specs, what Tog said last time and the last time it was made - this is a private input.

Reply:
I mean. The hash of the Note should be public. Otherways, how are you supposed to know that you're verifying indeed the Note that you think you are?

Reply:
As per a conversation from the standup, we have understood that the note_hash is private.